lib/commons/aria/get-role.js

```diff
@@ -152,4 +145,31 @@ function getRole(node, { noImplicit, ...explicitRoleOptions } = {}) {
 	return explicitRole;
 }
 
+/**
+ * Return the semantic role of an element.
+ *
+ * @method getRole
+ * @memberof axe.commons.aria
+ * @instance
+ * @param {Element|VirtualNode} node
+ * @param {Object} options
+ * @param {boolean} options.noImplicit  Do not return the implicit role // @deprecated
+ * @param {boolean} options.fallback  Allow fallback roles
+ * @param {boolean} options.abstracts  Allow role to be abstract
+ * @param {boolean} options.dpub  Allow role to be any (valid) doc-* roles
+ * @param {boolean} options.noPresentational return null if role is presentation or none
+ * @returns {string|null} Role or null
+ *
+ * @deprecated noImplicit option is deprecated. Use aria.getExplicitRole instead.
+ */
+function getRole(node, { noPresentational, ...options } = {}) {
+	const role = resolveRole(node, options);
+
+	if (noPresentational && ['presentation', 'none'].includes(role)) {
+		return null;
+	}
```

Comment:
I think this would be better done as an "else if" here https://github.com/dequelabs/axe-core/pull/2549/files#diff-1bf6cc2da8acf89f4e8559d8254862d7L143 instead of in a separate function. WDYT?

Reply:
Just to be sure I understand the suggestion clearly, I'll explain the reasoning behind my decisions.

I created the resolveRole(...) function from the understanding that the new option needs to handle _whatever role gets returned from getRole(...)_.  Since the original getRole(...) function had so many early returns, this was a great way for me to catch "whatever gets returned" as the role, and then apply the new option's behavior.

Let me know if that helps clarify why I did what I did.  Thank you @WilcoFiers for your review

Reply:
@Loganras You are absolutely right. @straker reminded me that implicitRole can return presentational too. My suggestion doesn't work. Let's go with this.